### ISSUE TITLE
[BUG] Picker - Cannot read properties of undefined (reading 'key') during keyboard navigation 

### ISSUE THREAD

**Issue by MateuszRomek** [livechat/design-system#820]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

During the filtering of items in picker, there is a problem with keyboard navigation, which causes a error in AA

### Steps To Reproduce

1. Open picker
2. Use keyboard to select random item
3. Select item with enter
4. Start filtering picker with search
5. Use arrow down and then arrow up 
6. Crash

### Expected Behavior

Shouldn't crash

### Screenshots

https://github.com/livechat/design-system/assets/46934668/9b34a39c-1e32-4d3f-af90-fee82fef8494

### Design system package(s) version

examples:
  - @livechat/design-system-react-components@1.0.0-beta.13

### Environment

```markdown
Chrome, Mac OS
```
